### PR TITLE
feat: Add `GuApiLambda` pattern

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Refactor the props to `GuLambdaFunction` to create an explicit pattern for an API Gateway executed lambda.

BREAKING CHANGE: Define an API Gateway executed lambda with the explicit `GuLambdaFunction` pattern

## Does this change require changes to existing projects or CDK CLI?

Yes, see notes above.

## How to test

See tests.

## How can we measure success?

The API to create a lambda is cleaner.

## Have we considered potential risks?

n/a

### CODE REVIEW

**Review by jacobwinch — APPROVED** [review]
Great work!

I wonder if this pattern should create an ACM certificate for the API (and eventually a CNAME for the API Gateway instance) too? If we decide it should, I think we could add that in a follow-up PR.

File: src/patterns/api-lambda.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+interface ApiProps extends Omit<LambdaRestApiProps, "handler"> {
+  id: string;
+}
+
+interface GuApiLambdaProps extends Omit<GuFunctionProps, "errorPercentageMonitoring"> {
+  apis: ApiProps[];
+  monitoringConfiguration: NoMonitoring | GuLambdaErrorPercentageMonitoringProps;
+}
+
+export class GuApiLambda extends GuLambdaFunction {
```
Comment (jacobwinch) [inline src/patterns/api-lambda.ts:17]: It's probably worth adding docs for this new pattern? Again, happy for that to follow in a separate PR if you'd prefer to get this merged.

**Comment by akash1810** [comment]
> I wonder if this pattern should create an ACM certificate for the API (and eventually a CNAME for the API Gateway instance) too? If we decide it should, I think we could add that in a follow-up PR.

Great idea and I think it makes sense to be its own PR.

**Comment by akash1810** [comment]
> I wonder if this pattern should create an ACM certificate for the API (and eventually a CNAME for the API Gateway instance) too? If we decide it should, I think we could add that in a follow-up PR.

Also thinking it would be good to add a CFN Output for API Gateway that results from this pattern to make DNS things a little easier. Will make that a small follow-up PR.

**Pull request merged**